src/ripple/app/ledger/LedgerTiming.h

```diff
@@ -20,42 +20,108 @@
 #ifndef RIPPLE_APP_LEDGER_LEDGERTIMING_H_INCLUDED
 #define RIPPLE_APP_LEDGER_LEDGERTIMING_H_INCLUDED
 
+#include <cstdint>
+#include <utility>
+
 namespace ripple {
 
+/** Determines whether the current ledger should close at this time.
+
+    This function should be called when a ledger is open and there is no close
+    in progress, or when a transaction is received and no close is in progress.
+
+    @param anyTransactions indicates whether any transactions have been received
+    @param previousProposers proposers in the last closing
+    @param proposersClosed proposers who have currently closed this ledger
+    @param proposersValidated proposers who have validated the last closed ledger
+    @param previousMSeconds time, in milliseconds, for the previous ledger to
+                            reach consensus (in milliseconds)
+    @param currentMSeconds time, in milliseconds since the previous ledger closed
+    @param openMSeconds time, in milliseconds, since the previous LCL was computed
+    @param idleInterval the network's desired idle interval
+*/
+bool shouldCloseLedger (
+    bool anyTransactions,
+    int previousProposers,      int proposersClosed,    int proposerersValidated,
+    int previousMSeconds,       int currentMSeconds,    int openMSeconds,
+    int idleInterval);
+
+/** Determine whether the network reached consensus and whether we joined.
+
+    @param previousProposers proposers in the last closing (not including us)
+    @param currentProposers proposers in this closing so far (not including us)
+    @param currentAgree proposers who agree with us
+    @param currentFinished proposers who have validated a ledger after this one
+    @param previousAgreeTime how long it took to agree on the last ledger
+    @param currentAgreeTime how long we've been trying to agree
+
+    @return a pair of booleans:
+        * If the first boolean is false we don't have consensus and the second
+          value is ignored.
+        * If the first boolean is true then we have consensus and the second
```

Comment:
There are three cases here, which maps poorly onto a pair of booleans.

I think you should do the following instead: `enum class Consensus {no, withUs, withoutUs};`

So 

`return {false, false};` -> `return Consensus::no;`
`return {true, false};` -> `return Consensus::withoutUs;`
`return {true, true};` -> `return Consensus::withUs;`

Isn't that more understandable?